docs/advanced_usages/scaling.md

```diff
@@ -0,0 +1,635 @@
+    - 💻 Open an issue on ANTA's [GitHub repository](https://github.com/aristanetworks/anta)
+    - 🤝 Join the discussion in our community channels
+
+    This guide also assumes you have a basic understanding of ANTA and are familiar with its core concepts. If you're new to ANTA, we recommend starting with the [ANTA Getting Started](../getting-started.md) before proceeding.
+
+ANTA (Arista Network Test Automation) is continually evolving to meet the demands of large-scale network validation. As networks grow in size and complexity, the need for efficient, scalable testing becomes increasingly important. This guide explores comprehensive strategies for scaling ANTA to handle large network fabrics effectively.
+
+We'll cover two main approaches:
+
+1. 🔄 Single Process Optimization: Leveraging ANTA's built-in asynchronous capabilities with resource management
+2. 🔀 Multi-Process Scaling: Running multiple ANTA instances in parallel (covered in [Part 2](#-multi-process-scaling))
+
+For large fabrics (500+ devices), you may need to combine both approaches for optimal performance.
+
+## 🔄 Single Process Optimization
+
+### Understanding ANTA's Asynchronous Nature
+
+ANTA uses Python's `asyncio` library for concurrent execution:
+
+- 📦 Each test is a coroutine in the main event loop
+- 🔄 Tests run concurrently, not in parallel
+- ⚙️ The event loop manages all coroutines and network I/O
+
+**Key Concept**: While `asyncio` provides excellent concurrency, there are practical limits to how many coroutines can be efficiently managed in a single event loop running on one CPU core. High coroutine counts increase event loop overhead, lead to frequent context switching, consume more memory, and may degrade performance.
+
+### Resource Management
+
+ANTA now provides several environment variables to control resource usage:
+
+!!! note "Default Values"
+    The values specified in the following examples are the actual default values used by ANTA. See the [Performance Tuning](#performance-tuning) section for guidance on adjusting these values.
+
+1. **Test Concurrency**:
+
+    ```bash
+    export ANTA_MAX_CONCURRENCY=10000
+    ```
+
+    This limits the number of concurrent test coroutines to prevent event loop overload. ANTA will schedule coroutines up to this limit, then wait for some to complete before scheduling more.
+
+2. **File Descriptors**:
+
+    ```bash
+    export ANTA_NOFILE=16384
+    ```
+
+    Sets the maximum number of open file descriptors for the ANTA process, usually for handling device connections. At startup, ANTA sets its process’s soft limit to the maximum allowed (up to `16384`). This adjustment is necessary because the soft limit is typically set to `1024`, while the hard limit is often higher (system-dependent). If ANTA’s hard limit is lower than the number of selected tests in ANTA, the process may request more file descriptors than the operating system allows, causing an error. In such cases, a WARNING is displayed at startup.
```

Comment:
```suggestion
    Sets the maximum number of open file descriptors for the ANTA process, usually for handling device connections. At startup, ANTA sets its process soft limit to the maximum allowed (up to `16384`). This adjustment is necessary because the soft limit is typically set to `1024`, while the hard limit is often higher (system-dependent). If ANTA hard limit is lower than the number of selected tests in ANTA, the process may request more file descriptors than the operating system allows, causing an error. In such cases, a WARNING is displayed at startup.
```

Also add a note that it does not work on Windows system today.
Maybve we can show the warning